blob/drivertest/drivertest.go

```diff
@@ -319,6 +321,219 @@ func testList(t *testing.T, newHarness HarnessMaker) {
 	}
 }
 
+type listResult struct {
+	Key   string
+	IsDir bool
+	Sub   []*listResult
+}
+
+func (r *listResult) String() string {
+	if r.IsDir {
+		return fmt.Sprintf("Dir %s (%d entries)", r.Key, len(r.Sub))
+	}
+	return r.Key
+}
+
+// listRecursively recursively lists b using prefix and delim.
+func listRecursively(ctx context.Context, b *blob.Bucket, prefix, delim string) ([]*listResult, error) {
+	iter, err := b.List(ctx, &blob.ListOptions{
+		Prefix:    prefix,
+		Delimiter: delim,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var retval []*listResult
+	for {
+		obj, err := iter.Next(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if obj == nil {
```

Comment:
I think when we adopt an error sentinel for Next, this case will go away (become impossible).

Reply:
Agree.